hack/envtest/build.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -x
+
+# This script is used to build the test binaries for the envtest package.
+# It's intended to be run from the root of the repository, through the Makefile.
+
+envtest_dir=$(dirname ${BASH_SOURCE})
+ROOT=$(dirname "${BASH_SOURCE[0]}")/../../..
+
+# Check modified files in hack/envtest/_matrix/*.yaml
+# If there are any changes, we need to rebuild the test binaries.
+# This is to ensure that the test binaries are always up-to-date with the matrix files.
+
+changed_files=$(git diff --name-only --relative hack/envtest/_matrix | grep -E '\.yaml$' || true)
```

Comment:
In which case do we need the `|| true`?

Reply:
It'll fail with an error when no yaml greps are found, so I added the true, and it should exit cleanly

Reply:
or maybe not? Can't remember if that's what I saw locally or not

Reply:
Sounds fine